### ISSUE TITLE
Regression: strange behaviour of indent rules in v0.43.2

### ISSUE THREAD

**Issue by m-burst** [pinterest/ktlint#1330]
## Expected Behavior

I've tried to upgrade from v0.40.0 to v0.43.2 and encountered a situation where the indent rule reports some strange violations.

Below is a minimal reproducible example of the issue via CLI:

```bash
ktlint --stdin <<EOF
fun func(lambdaArg: Unit.() -> Unit = { }, secondArg: Int) {
    println()
}
EOF
```

I'm expecting this code to be valid.

## Observed Behavior

ktlint reports the following errors with the code above:

```
<stdin>:2:1: Unexpected indentation (4) (should be 0)
<stdin>:3:1: Unexpected indentation (0) (should be -4)
```

## Steps to Reproduce

See code above

## Your Environment

* Version of ktlint used: 0.43.2
* Name and version (or code for custom task) of integration used (Gradle plugin, Maven plugin, command line, custom Gradle task): cli
* Version of Gradle used (if applicable): n/a
* Operating System and version: 
* Link to your project (if it's a public repository): n/a
